modules/images/dominant-color/load.php

```diff
@@ -126,8 +126,12 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $
 		$extra_class  = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
 	}
 
-	if ( ! empty( $data ) || ! empty( $style ) ) {
-		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+	if ( ! empty( $data ) || ! empty( $style ) || ! str_contains( $data, $filtered_image ) ) {
+		if ( str_contains( $filtered_image, 'style=' ) ) {
+			$filtered_image = str_replace( 'style="', $data . 'style="' . $style, $filtered_image );
```

Comment:
We need to make sure this actually looks at the `style` attribute. Right now it would also match something like `data-style`, which we don't want.
```suggestion
		if ( str_contains( $filtered_image, ' style=' ) ) {
			$filtered_image = str_replace( ' style="', ' ' . $data . 'style="' . $style, $filtered_image );
```